Ticket#: 2010111510003474
Increase dhcp-timeout from 10 to 60 seconds to work-around default "bridge forwarding delay" of 15 seconds for xen instances.
Dummy-change to Config is needed to update the time-stamp on Config to force make to remake etherboot.
--- a/tools/firmware/Makefile	2010-11-17 08:35:01.000000000 +0100
+++ b/tools/firmware/Makefile	2010-11-17 08:35:07.000000000 +0100
@@ -8,7 +8,7 @@
 SUBDIRS :=
 SUBDIRS += rombios
 SUBDIRS += vgabios
-#SUBDIRS += etherboot
+SUBDIRS += etherboot
 SUBDIRS += hvmloader
 
 .PHONY: all
--- a/tools/firmware/etherboot/patches/series	2010-11-17 08:34:41.000000000 +0100
+++ b/tools/firmware/etherboot/patches/series	2010-11-17 08:34:49.000000000 +0100
@@ -1 +1,3 @@
 boot_prompt_option.patch
+dhcp_timeout.patch
+clean_config.patch
--- a/tools/firmware/etherboot/patches/dhcp_timeout.patch	2010-11-17 08:34:14.000000000 +0100
+++ b/tools/firmware/etherboot/patches/dhcp_timeout.patch	2010-11-17 08:34:14.000000000 +0100
@@ -0,0 +1,11 @@
+--- gpxe/src/include/gpxe/dhcp.h.orig	2010-11-17 08:33:53.000000000 +0100
++++ gpxe/src/include/gpxe/dhcp.h	2010-11-17 08:19:52.000000000 +0100
+@@ -469,7 +469,7 @@
+ 
+ /** Timeouts for sending DHCP packets */
+ #define DHCP_MIN_TIMEOUT ( 1 * TICKS_PER_SEC )
+-#define DHCP_MAX_TIMEOUT ( 10 * TICKS_PER_SEC )
++#define DHCP_MAX_TIMEOUT ( 60 * TICKS_PER_SEC )
+ 
+ /** Settings block name used for DHCP responses */
+ #define DHCP_SETTINGS_NAME "dhcp"
--- a/tools/firmware/etherboot/patches/clean_config.patch	2010-11-19 12:31:14.000000000 +0100
+++ b/tools/firmware/etherboot/patches/clean_config.patch	2010-11-19 12:31:14.000000000 +0100
@@ -0,0 +1,9 @@
+--- gpxe/src/Makefile.orig	2010-11-19 12:31:55.000000000 +0100
++++ gpxe/src/Makefile	2010-11-19 12:31:44.000000000 +0100
+@@ -182,4 +182,6 @@
+ 
+ # Drag in the automatic build system and other housekeeping functions
+ MAKEDEPS	+= Makefile.housekeeping
++MAKEDEPS	+= config/general.h
++serial_DEPS := config/serial.h
+ include Makefile.housekeeping
--- a/tools/firmware/etherboot/Makefile        2010-11-19 16:37:52.000000000 +0100
+++ b/tools/firmware/etherboot/Makefile     2010-11-19 16:36:12.000000000 +0100
@@ -3,6 +3,10 @@
 XEN_ROOT = ../../..
 include $(XEN_ROOT)/tools/Rules.mk
 include Config
+# Xens make parameters are incompatible with Etherboot
+CFLAGS :=
+HOSTCFLAGS :=
+LDFLAGS :=
 
 D=gpxe
 T=gpxe-git-snapshot.tar.gz
--- a/tools/firmware/etherboot/Config	2010-11-17 08:39:21.000000000 +0100
+++ b/tools/firmware/etherboot/Config	2010-11-17 08:39:32.000000000 +0100
@@ -1,4 +1,3 @@
-
 NICS = rtl8139 eepro100
 
 CFLAGS += -UPXE_DHCP_STRICT
